sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/tests/JObjectTests.cs

```diff
@@ -372,25 +404,202 @@ public void TestWebPubSubContext_UserEventStates()
+                connectionStates: states);
+            var test = new WebPubSubContext(new UserEventRequest(context, BinaryData.FromString("test"), WebPubSubDataType.Text));
+
+            var jObj = JObject.Parse(JsonConvert.SerializeObject(test));
+            var request = jObj["request"];
+            Assert.NotNull(request);
+            Assert.AreEqual("test", request["data"].Value<string>());
+            Assert.NotNull(jObj["response"]);
+            Assert.AreEqual(null, jObj["errorMessage"].Value<string>());
+            Assert.AreEqual("False", jObj["hasError"].Value<string>());
+            Assert.AreEqual("False", jObj["isPreflight"].Value<string>());
+            var context1 = request["connectionContext"];
+            Assert.NotNull(context1);
+            var states1 = context1["states"];
+            Assert.NotNull(states1);
+            Assert.AreEqual("aValue", states1["aKey"].Value<string>());
+            Assert.AreEqual(123, states1["bKey"].Value<int>());
+            var classStates = states1["cKey"];
+            Assert.NotNull(classStates);
+            Assert.AreEqual("GA", classStates["Title"].Value<string>());
+        }
+
+        [TestCase]
+        public void TestWebPubSubContext_UserEventStates_AllBinaryData()
+        {
+            var ctx = new WebPubSubConnectionContext(
+                eventType: WebPubSubEventType.System,
+                eventName: "connected",
+                hub: null,
+                connectionId: "connectionId",
+                userId: "userA",
+                signature: null,
+                origin: null,
+                states: new Dictionary<string, object>
+                {
+                    { "aKey", "aValue"},
+                    { "bKey", 123 },
+                    { "cKey", new StateTestClass(DateTime.Now, "Test", 42) }
+                },
+                headers: null);
+
+            Assert.IsInstanceOf<BinaryData>(ctx.ConnectionStates["aKey"]);
+            Assert.IsInstanceOf<string>(ctx.States["aKey"]);
+            Assert.AreEqual("aValue", ctx.ConnectionStates["aKey"].ToObjectFromJson<string>());
+            Assert.AreEqual("\"aValue\"", ctx.States["aKey"]);
+
+            Assert.IsInstanceOf<BinaryData>(ctx.ConnectionStates["bKey"]);
+            Assert.IsInstanceOf<string>(ctx.States["bKey"]);
+            Assert.AreEqual(123, ctx.ConnectionStates["bKey"].ToObjectFromJson<int>());
+            Assert.AreEqual("123", ctx.States["bKey"]);
+
+            Assert.IsInstanceOf<BinaryData>(ctx.ConnectionStates["cKey"]);
+            Assert.IsInstanceOf<string>(ctx.States["cKey"]);
+            Assert.AreEqual(42, ctx.ConnectionStates["cKey"].ToObjectFromJson<StateTestClass>().Version);
+            StringAssert.StartsWith("{", (string)ctx.States["cKey"]);
+        }
+
+        [TestCase]
+        public void TestWebPubSubContext_UserEventStates_NotDoubleSerialized()
+        {
+            WebPubSubConfigProvider.RegisterJsonConverter();
```

Comment:
please note that this changes the global config so it might have a side impact on other UTs who do not expect such converters

Reply:
Move to class ctor. It can be viewed as always expected in the functions tests. Some tests didn't call due to not hit those convertors.